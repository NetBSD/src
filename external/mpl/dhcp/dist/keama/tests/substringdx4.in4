# substring data expression

# empty configs are not accepted by Kea
default-lease-time 1800;

# use substring in a reductible match
class "reductible" {
    match substring(option host-name, 0, 3);
}

subclass "reductible" "www" { }

# reduce literals too
class "literal" {
    match if option host-name = substring("www.example.com", 0, 3);
}

# raw
option domain-name = substring(option domain-name, 4, 1000);


