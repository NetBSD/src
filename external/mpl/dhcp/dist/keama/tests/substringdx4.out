{
  # substring data expression
  # empty configs are not accepted by Kea
  "Dhcp4": {
    "valid-lifetime": 1800,
    "client-classes": [
      # use substring in a reductible match
      /// match: substring(option dhcp.host-name, 0, 3)
      {
        "name": "reductible"
      },
      /// subclass selector 'www'
      {
        "name": "sub#reductible#0",
        /// from: match substring(option dhcp.host-name, 0, 3)
        /// data: 'www'
        "test": "substring(option[12].hex,0,3) == 'www'"
      },
      # reduce literals too
      {
        "name": "literal",
        /// from: match if (option dhcp.host-name) = (substring('www.example.com', 0, 3))
        "test": "option[12].hex == 'www'"
      }
    ],
    "option-data": [
//    # raw
//    {
//      "space": "dhcp4",
//      "name": "domain-name",
//      "code": 15,
//      "csv-format": false,
//      "expression": {
//        "substring": {
//          "expression": {
//            "option": {
//              "universe": "dhcp",
//              "name": "domain-name",
//              "code": 15
//            }
//          },
//          "offset": 4,
//          "length": 1000
//        }
//      }
//    }
    ]
  }
}
